src/main/java/gregtech/core/unification/material/internal/MaterialRegistryManager.java

```diff
@@ -0,0 +1,144 @@
+    }
+
+    @Nonnull
+    @Override
+    public MaterialRegistry getRegistry(@Nonnull String modid) {
+        MaterialRegistry registry = registries.get(modid);
+        return registry != null ? registry : gregtechRegistry;
+    }
+
+    @Nonnull
+    @Override
+    public MaterialRegistry getRegistry(int networkId) {
+        MaterialRegistry registry = networkIds.get(networkId);
+        return registry != null ? registry : gregtechRegistry;
+    }
+
+    @Nonnull
+    @Override
+    public Collection<MaterialRegistry> getRegistries() {
+        if (getPhase() == Phase.PRE) {
+            throw new IllegalStateException("Cannot get all material registries during phase " + getPhase());
+        }
+        return Collections.unmodifiableCollection(registries.values());
+    }
+
+    @Nonnull
+    @Override
+    public Collection<Material> getRegisteredMaterials() {
+        if (registeredMaterials == null ||
+                (getPhase() != Phase.CLOSED && getPhase() != Phase.FROZEN)) {
+            throw new IllegalStateException("Cannot retrieve all materials before registration");
+        }
+        return registeredMaterials;
+    }
+
+    @Override
+    public Material getMaterial(String name) {
+        if (!name.isEmpty()) {
+            String modid;
+            String materialName;
+            int index = name.indexOf(':');
+            if (index >= 0) {
+                modid = name.substring(0, index);
+                materialName = name.substring(index + 1);
+            } else {
+                modid = GTValues.MODID;
+                materialName = name;
+            }
+            return getRegistry(modid).getObject(materialName);
+        }
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public Phase getPhase() {
+        return registrationPhase;
+    }
+
+    public void unfreezeRegistries() {
+        registries.values().forEach(MaterialRegistryImpl::unfreeze);
+        registrationPhase = Phase.OPEN;
+    }
+
+    public void closeRegistries() {
+        registries.values().forEach(MaterialRegistryImpl::closeRegistry);
+        Collection<Material> collection = new ArrayList<>();
+        for (MaterialRegistry registry : registries.values()) {
+            collection.addAll(registry.getAllMaterials());
+        }
+        registeredMaterials = Collections.unmodifiableCollection(collection);
+        Material.Builder.setConstructionRegistry(null);
+        registrationPhase = Phase.CLOSED;
+    }
+
+    public void freezeRegistries() {
+        registries.values().forEach(MaterialRegistryImpl::freeze);
+        registrationPhase = Phase.FROZEN;
+    }
+
+    private MaterialRegistryImpl createInternalRegistry() {
+        MaterialRegistryImpl registry = new MaterialRegistryImpl(GTValues.MODID);
+        this.registries.put(GTValues.MODID, registry);
+        return registry;
+    }
+
+    protected Material getDefaultFallback() {
+        return gregtechRegistry.getFallbackMaterial();
+    }
```

Comment:
This seems like somewhat of a loop, going to MaterialRegistryImpl, and then back here to MaterialRegistryManager. Also, as far as I can tell we never set the default fallback material for `gregtechRegistry`